### ISSUE TITLE
Enhancement: DxDataGrid should provide a way to bind to a remote data source

### ISSUE THREAD

**Issue by drinkwater99** [DevExpress/Blazor#59]
According to the docs the grid only works with an IEnumerable or IQueryable data source.
How can you bind the grid when calling a Web Api without selecting the whole data set ?

**Comment by akorenchikov** [comment]
Hi @drinkwater99 ,

Currently, we are working on this feature.
Could you share your remote binding scenario so that we will take it into account and cover more cases of this feature's usage?

**Comment by drinkwater99** [comment]
I think the grid would need to have a OnGetData event which would fire whenever data is needed (paging, filtering, sorting...). 
The arguments for that event would provide the needed info (page number, current filter conditions, current sort field(s)...) to be able query the backend data and return a subset of the data (as well as the total row count). We would simply reinitialise the grid's Data property variable with a new IEnumerable which would contain only a subset of the data.

**Comment by zabelin-vladimir** [comment]
Hello @drinkwater99

I'm happy to announce that we implemented the [CustomData](https://docs.devexpress.com/Blazor/DevExpress.Blazor.Base.DxDataGridBase-1.CustomData) property for DxDataGrid in our [latest update (19.2 Beta)](https://community.devexpress.com/blogs/aspnet/archive/2020/01/17/blazor-components-localization-asynchronous-data-operations-and-more-v19-2-0.aspx).

Check the [Remote Data Source](https://demos.devexpress.com/Blazor/GridRemoteDataSource#CustomDataSource) demo which illustrates how to use this property with our DataGrid and DxDataGridComboBoxColumn column.

I will be glad to hear your feedback about these events and other features which we implemented.
